src/System.Drawing.Common/src/System/Drawing/Icon.Windows.cs

```diff
@@ -374,7 +374,46 @@ private void DrawIcon(IntPtr dc, Rectangle imageRect, Rectangle targetRect, bool
             }
             finally
             {
-                SafeNativeMethods.RestoreClipRgn(dc, hSaveRgn);
+                RestoreClipRgn(dc, hSaveRgn);
+            }
+        }
+
+        private static IntPtr SaveClipRgn(IntPtr hDC)
+        {
+            IntPtr hTempRgn = Interop.Gdi32.CreateRectRgn(0, 0, 0, 0);
+            IntPtr hSaveRgn = IntPtr.Zero;
+            try
+            {
+                int result = Interop.Gdi32.GetClipRgn(hDC, hTempRgn);
+                if (result > 0)
+                {
+                    hSaveRgn = hTempRgn;
+                    hTempRgn = IntPtr.Zero;
+                }
+            }
+            finally
+            {
+                if (hTempRgn != IntPtr.Zero)
+                {
+                    Interop.Gdi32.DeleteObject(hTempRgn);
+                }
+            }
+
+            return hSaveRgn;
+        }
+
+        private static void RestoreClipRgn(IntPtr hDC, IntPtr hRgn)
+        {
+            try
+            {
+                Interop.Gdi32.SelectClipRgn(new HandleRef(null, hDC), new HandleRef(null, hRgn));
```

Comment:
`HandleRef` with a null is pointless. The try block doesn't help here either as it shouldn't throw "normally" (see the comment above).

If these methods are only called in one place it might be better to move them inline.